graphql_compiler/schema_generation/orientdb/schema_graph_builder.py

```diff
@@ -448,3 +468,67 @@ def _get_graphql_representation_of_non_graph_elements(non_graph_elements, inheri
             }
             graphql_reps[element_name] = GraphQLObjectType(element_name, fields, [])
     return graphql_reps
+
+
+def _get_indexes(index_data, elements):
+    """Return a set of IndexDefinitions describing the indexes defined in the OrientDB database."""
+    all_indexes = set()
+
+    # Get indexes from OrientDB.
+    all_class_names = set(elements.keys())
+
+    for index in index_data:
+        index_name = index['name']
+        index_type = index['type']
+        index_unique = index_type in UNIQUE_INDEX_TYPES
+        index_ordered = index_type in ORDERED_INDEX_TYPES
+        index_definition = index['indexDefinition']
+        index_base_classname = index_definition['className']
+        index_ignore_nulls = index_definition['nullValuesIgnored']
+
+        # Exclude indexes on non-graph classes (e.g. OUser).
+        if index_base_classname not in all_class_names:
+            continue
+
+        is_edge_index = isinstance(elements[index_base_classname], EdgeType)
+        if is_edge_index:
+            # OrientDB sometimes decides to make edge indexes ignore nulls.
+            # This is not something we want to allow, hard-code it to False.
+            # While this makes us diverge from OrientDB's schema,
+            # it's safe to be *more strict* than OrientDB.
+            index_ignore_nulls = False
```

Comment:
I'm not sure if we should do this. Looks like it's Kensho business logic, and can be fixed in the `index_data` input dict. But also, it might be a sane thing to do for any user of this library, given that we document it.

Reply:
Yeah it's a good question. I think I originally wrote this code when I realized that OrientDB across different versions in the 2.x series had inconsistent defaults on whether nulls are allowed in unique indexes that apply to edges. In general, though, `null` values on edge endpoints don't make sense -- they would imply that you could have edges with only one endpoint, or worse, no endpoints.

So overall, I am leaning toward leaving it in, but I am open to being convinced otherwise.

Reply:
but this only relates to edge endpoints. It still makes sense to ignore nulls on other edge properties, no? Since we already assert you can only ignore nulls if the index is defined on a single column, we should at least check if that column is an endpoint, and only apply the rule in that case.

Reply:
There are some subtleties of OrientDB's index description format that we are glossing over, but I don't think it's worth going into them. Let's just remove this entire block, since it's the definition of legacy code :)